tests/unit/gpu_av_buffer_device_address.cpp

```diff
@@ -834,45 +833,115 @@ TEST_F(NegativeGpuAVBufferDeviceAddress, StoreRelaxedBlockLayout) {
+       %ufoo = OpTypeStruct %_ptr_PhysicalStorageBuffer_bufStruct
+      %float = OpTypeFloat 32
+    %v3float = OpTypeVector %float 3
+  %bufStruct = OpTypeStruct %float %v3float
+%_ptr_PhysicalStorageBuffer_bufStruct = OpTypePointer PhysicalStorageBuffer %bufStruct
+%_ptr_Uniform_ufoo = OpTypePointer Uniform %ufoo
+       %ssbo = OpVariable %_ptr_Uniform_ufoo Uniform
+        %int = OpTypeInt 32 1
+      %int_0 = OpConstant %int 0
+%_ptr_Uniform__ptr_PhysicalStorageBuffer_bufStruct = OpTypePointer Uniform %_ptr_PhysicalStorageBuffer_bufStruct
+   %float_42 = OpConstant %float 42
+%_ptr_PhysicalStorageBuffer_float = OpTypePointer PhysicalStorageBuffer %float
+       %main = OpFunction %void None %3
+          %5 = OpLabel
+         %16 = OpAccessChain %_ptr_Uniform__ptr_PhysicalStorageBuffer_bufStruct %ssbo %int_0
+         %17 = OpLoad %_ptr_PhysicalStorageBuffer_bufStruct %16
+         %20 = OpAccessChain %_ptr_PhysicalStorageBuffer_float %17 %int_0
+               OpStore %20 %float_42 Aligned 16
+               OpReturn
+               OpFunctionEnd
+    )";
+
+    // Make a uniform buffer to be passed to the shader that contains the pointer
+    const uint32_t uniform_buffer_size = 8;  // 64 bits pointer
+    VkMemoryPropertyFlags mem_props = VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT | VK_MEMORY_PROPERTY_HOST_COHERENT_BIT;
+    vkt::Buffer uniform_buffer(*m_device, uniform_buffer_size, VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT, mem_props);
+
+    CreateComputePipelineHelper pipeline(*this);
+    pipeline.cs_ =
+        std::make_unique<VkShaderObj>(this, shader_source, VK_SHADER_STAGE_COMPUTE_BIT, SPV_ENV_VULKAN_1_2, SPV_SOURCE_ASM);
+    pipeline.dsl_bindings_ = {{0, VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER, 1, VK_SHADER_STAGE_COMPUTE_BIT, nullptr}};
+    pipeline.CreateComputePipeline();
+
+    pipeline.descriptor_set_->WriteDescriptorBufferInfo(0, uniform_buffer, 0, VK_WHOLE_SIZE);
+    pipeline.descriptor_set_->UpdateDescriptorSets();
+
+    m_command_buffer.Begin();
+    vk::CmdBindPipeline(m_command_buffer, VK_PIPELINE_BIND_POINT_COMPUTE, pipeline.Handle());
+    vk::CmdBindDescriptorSets(m_command_buffer, VK_PIPELINE_BIND_POINT_COMPUTE, pipeline.pipeline_layout_.handle(), 0, 1,
+                              &pipeline.descriptor_set_->set_, 0, nullptr);
+    vk::CmdDispatch(m_command_buffer, 1, 1, 1);
+    m_command_buffer.End();
+
+    const uint32_t storage_buffer_size = 3 * sizeof(float);  // only can fit 3 floats
+    vkt::Buffer storage_buffer(*m_device, storage_buffer_size, 0, vkt::device_address);
+    const VkDeviceAddress storage_buffer_addr = storage_buffer.Address();
+
+    // Base buffer address is (storage_buffer_addr - 16), so expect writing to `f` to cause an OOB access
+    auto data = static_cast<VkDeviceAddress *>(uniform_buffer.Memory().Map());
+    // The OpStore is aligned to 16 bytes, so need to substract by that
+    data[0] = storage_buffer_addr - 16;
```

Comment:
```suggestion
    uniform_buffer_ptr[0] = storage_buffer_addr - 16;
```
I will erase `data` from your vocabulary, whatever it costs (╯°□°）╯︵ ┻━┻

Reply:
updated all the other locations in the BDA tests